docs/man/scalac.1
.\" ##########################################################################
.\" #                      __                                                #
.\" #      ________ ___   / /  ___     Scala 2 On-line Manual Pages          #
.\" #     / __/ __// _ | / /  / _ |    (c) 2002-2007, LAMP/EPFL              #
.\" #   __\ \/ /__/ __ |/ /__/ __ |                                          #
.\" #  /____/\___/_/ |_/____/_/ | |    http://scala-lang.org/                #
.\" #                           |/                                           #
.\" ##########################################################################
.\"
.\" Process this file with nroff -man scala.1
.\"
.TH scalac 1  "April 18, 2007" "version 0.4" "USER COMMANDS"
.\"
.\" ############################## NAME ###############################
.\"
.SH NAME
.PP
\fBscalac\fR \- Compiler for the Scala 2 language
.\"
.\" ############################## SYNOPSIS ###############################
.\"
.SH SYNOPSIS
.PP
\fBscalac\fR  [ <\fIoptions\fR> ] <\fIsource files\fR>
.\"
.\" ############################## PARAMETERS ###############################
.\"
.SH PARAMETERS
.PP
.TP
<\fIoptions\fR>
Command line options. See \fBOPTIONS\fR below.
.TP
<\fIsource files\fR>
One or more source files to be compiled (such as MyClass.scala).
.\"
.\" ############################## OPTIONS ###############################
.\"
.SH OPTIONS
.PP
The compiler has a set of standard options that are supported on the current development environment and will be supported in future releases. An additional set of non-standard options are specific to the current virtual machine implementation and are subject to change in the future.  Non-standard options begin with \fB-X\fR.
.\"
.\" ############################## Standard Options ###############################
.\"
.SS "Standard Options"
.PP
.TP
\fB\-g:{none,source,line,vars,notc}\fR 
"none" generates no debugging info,
.IP
"source" generates only the source file attribute,
.IP
"line" generates source and line number information,
.IP
"vars" generates source, line number and local variable information,
.IP
"notc" generates all of the above and \fIwill not\fR perform tail call optimization.
.IP

.TP
\fB\-nowarn\fR 
Generate no warnings
.TP
\fB\-verbose\fR 
Output messages about what the compiler is doing
.TP
\fB\-deprecation\fR 
Indicate whether source should be compiled with deprecation information; defaults to off (accepted values are: on, off, yes and no)
.IP
Available since Scala version 2.2.1
.IP

.TP
\fB\-unchecked\fR 
Enable detailed unchecked warnings
.IP
Non variable type-arguments in type patterns are unchecked since they are eliminated by erasure
.IP
Available since Scala version 2.3.0
.IP

.TP
\fB\-classpath\fR <\fIpath\fR> 
Specify where to find user class files (on Unix-based systems a colon-separated list of paths, on Windows-based systems, a semicolon-separate list of paths). This does not override the built-in ("boot") search path.
.IP
The default class path is the current directory. Setting the CLASSPATH variable or using the -classpath command-line option overrides that default, so if you want to include the current directory in the search path, you must include "." in the new settings.
.IP

.TP
\fB\-sourcepath\fR <\fIpath\fR> 
Specify where to find input source files.
.TP
\fB\-bootclasspath\fR <\fIpath\fR> 
Override location of bootstrap class files (where to find the standard built-in classes, such as "scala.List").
.TP
\fB\-extdirs\fR <\fIdirs\fR> 
Override location of installed extensions.
.TP
\fB\-d\fR <\fIdirectory\fR> 
Specify where to place generated class files.
.TP
\fB\-encoding\fR <\fIencoding\fR> 
Specify character encoding used by source files.
.IP
The default value is platform-specific (Linux: "UTF8", Windows: "Cp1252"). Executing the following code in the Scala interpreter will return the default value on your system:
.IP
\fB    scala> \fRnew java.io.InputStreamReader(System.in).getEncoding
.IP

.TP
\fB\-target:\fR <\fItarget\fR> 
Specify which backend to use (jvm-1.5,jvm-1.4,msil,cldc).
.IP
The default value is "jvm-1.5" (was "jvm-1.4" up to Scala version 2.6.1).
.IP

.TP
\fB\-print\fR 
Print program with all Scala-specific features removed
.TP
\fB\-optimise\fR 
Generates faster bytecode by applying optimisations to the program
.TP
\fB\-explaintypes\fR 
Explain type errors in more detail.
.TP
\fB\-uniqid\fR 
Print identifiers with unique names (debugging option).
.TP
\fB\-version\fR 
Print product version and exit.
.TP
\fB\-help\fR 
Print a synopsis of standard options.
.\"
.\" ############################## Advanced Options ###############################
.\"
.SS "Advanced Options"
.PP
.TP
\fB\-Xassem\fR <\fIfile\fR> 
Name of the output assembly (only relevant with -target:msil)
.TP
\fB\-Xassem-path\fR <\fIpath\fR> 
List of assemblies referenced by the program (only relevant with -target:msil)
.TP
\fB\-Xcheck-null\fR 
Emit warning on selection of nullable reference
.TP
\fB\-Xdisable-assertions\fR 
Generate no assertions and assumptions
.TP
\fB\-Xexperimental\fR 
enable experimental extensions
.TP
\fB\-Xno-uescape\fR 
Disable handling of \eu unicode escapes
.TP
\fB\-Xplug-types\fR 
Parse but ignore annotations in more locations
.TP
\fB\-Xplugin:\fR <\fIfile\fR> 
Load a plugin from a file
.TP
\fB\-Xplugin-disable:\fR <\fIplugin\fR> 
Disable a plugin
.TP
\fB\-Xplugin-list\fR 
Print a synopsis of loaded plugins
.TP
\fB\-Xplugin-opt:\fR <\fIplugin:opt\fR> 
Pass an option to a plugin
.TP
\fB\-Xplugin-require:\fR <\fIplugin\fR> 
Abort unless a plugin is available
.TP
\fB\-Xprint:\fR <\fIphases\fR> 
Print out program after <\fIphases\fR> (see below).
.TP
\fB\-Xprint-pos\fR 
Print tree positions (as offsets)
.TP
\fB\-Xprint-types\fR 
Print tree types (debugging option).
.TP
\fB\-Xprompt\fR 
Display a prompt after each error (debugging option).
.TP
\fB\-Xresident\fR 
Compiler stays resident, files to compile are read from standard input.
.TP
\fB\-Xshow-class\fR <\fIclass\fR> 
Show class info.
.TP
\fB\-Xshow-object\fR <\fIobject\fR> 
Show object info.
.TP
\fB\-Xshow-phases\fR 
Print a synopsis of compiler phases.
.TP
\fB\-Xsource-reader\fR <\fIclassname\fR> 
Specify a custom method for reading source files.
.TP
\fB\-Xscript\fR <\fIobject\fR> 
Compile as a script, wrapping the code into object.main().
.\"
.\" ############################## Compilation Phases ###############################
.\"
.SS "Compilation Phases"
.PP
.TP
\fIinitial\fR
initializing compiler
.TP
\fIparse\fR
parse source files
.TP
\fInamer\fR
create symbols
.TP
\fIanalyze\fR
name and type analysis
.TP
\fIrefcheck\fR
reference checking
.TP
\fIuncurry\fR
uncurry function types and applications
.TP
\fItransmatch\fR
translate match expressions
.TP
\fIlambdalift\fR
lambda lifter
.TP
\fItypesasvalues\fR
represent types as values
.TP
\fIaddaccessors\fR
add accessors for constructor arguments
.TP
\fIexplicitouterclasses\fR
make links from inner classes to enclosing one explicit
.TP
\fIaddconstructors\fR
add explicit constructor for each class
.TP
\fItailcall\fR
add tail-calls
.TP
\fIwholeprog\fR
perform whole program analysis
.TP
\fIaddinterfaces\fR
add one interface per class
.TP
\fIexpandmixins\fR
expand mixins by code copying
.TP
\fIboxing\fR
makes boxing explicit
.TP
\fIerasure\fR
type eraser
.TP
\fIicode\fR
generate icode
.TP
\fIcodegen\fR
enable code generation
.TP
\fIterminal\fR
compilation terminated
.TP
\fIall\fR
matches all phases
.\"
.\" ############################## ENVIRONMENT ###############################
.\"
.SH ENVIRONMENT
.PP
.TP
\fBJAVACMD\fR
Specify the \fBjava\fR command to be used for running the Scala code.  Arguments may be specified as part of the environment variable; spaces, quotation marks, etc., will be passed directly to the shell for expansion.
.TP
\fBJAVA_OPTS\fR
Specify the options to be passed to the \fBjava\fR command defined by \fBJAVACMD\fR.
.IP
With Java 1.5 (or newer) one may for example configure the memory usage of the JVM as follows: JAVA_OPTS="-Xmx512M -Xms16M -Xss16M"
.IP
With GNU Java one may configure the memory usage of the GIJ as follows: JAVA_OPTS="--mx512m --ms16m"
.IP

.\"
.\" ############################## EXAMPLES ###############################
.\"
.SH EXAMPLES
.PP
.TP
Compile a Scala program to the current directory
\fBscalac\fR HelloWorld
.TP
Compile a Scala program to the destination directory \fBclasses\fR
\fBscalac\fR \fB\-d\fR classes HelloWorld.scala
.TP
Compile a Scala program using a user-defined \fBjava\fR command
\fBenv JAVACMD\fR=/usr/local/bin/cacao \fBscalac\fR \fB\-d\fR classes HelloWorld.scala
.TP
Compile all Scala files found in the source directory \fBsrc\fR to the destination directory \fBclasses\fR
\fBscalac\fR \fB\-d\fR classes src/*.scala
.\"
.\" ############################## EXIT STATUS ###############################
.\"
.SH "EXIT STATUS"
.PP
\fBscalac\fR returns a zero exist status if it succeeds to compile the specified input files. Non zero is returned in case of failure.
.\"
.\" ############################## AUTHOR ###############################
.\"
.SH AUTHOR
.PP
Written by Martin Odersky and other members of the Scala team.
.\"
.\" ############################## REPORTING BUGS ###############################
.\"
.SH "REPORTING BUGS"
.PP
Report bugs to <scala@listes.epfl.ch>.
.\"
.\" ############################## COPYRIGHT ###############################
.\"
.SH COPYRIGHT
.PP
This is open-source software, available to you under a BSD-like license. See accomponying "copyright" or "LICENSE" file for copying conditions. There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
.\"
.\" ############################## SEE ALSO ###############################
.\"
.SH "SEE ALSO"
.PP
\fBfsc\fR(1), \fBsbaz\fR(1), \fBscala\fR(1), \fBscaladoc\fR(1), \fBscalap\fR(1)


debian/changelog
scala (2.6.1-1) unstable; urgency=low

  * New upstream release

 -- Lex Spoon <lex@debian.org>  Tue, 15 Jan 2008 18:08:31 +0100

scala (2.6.0-1) unstable; urgency=low

  * New upstream release

 -- Lex Spoon <lex@debian.org>  Fri, 14 Sep 2007 15:22:05 +0100

scala (2.5.0-1) unstable; urgency=low

  * New upstream release
  * Depend only on java-gcj-compat, not java-gcj-compat-dev

 -- Lex Spoon <lex@debian.org>  Tue, 22 May 2007 16:51:47 +0200

scala (2.4.0-1) unstable; urgency=low

  * New upstream version.

  * Deleting all mention of the Sun JVM, just because some
    DD's find it bothersome.

  * Deleted mention of JDK5-specific classes, while awaiting
    the Sun JVM to become open source.  (Closes: #409785)

  * Added java-gcj-compat-dev as an install dependency.  (Closes: #408353)

 -- Lex Spoon <lex@debian.org>  Sun, 25 Mar 2007 11:56:18 +0200

scala (2.3.2-1) unstable; urgency=low

  * New upstream version

 -- Lex Spoon <lex@debian.org>  Fri, 19 Jan 2007 14:45:51 +0100

scala (2.3.0-1) unstable; urgency=low

  * New upstream version
  * Added a Java compiler as a build-dependency

 -- Lex Spoon <lex@debian.org>  Fri, 24 Nov 2006 16:02:09 +0100

scala (2.2.0-1) unstable; urgency=low

  * New upstream release

 -- Lex Spoon <lex@debian.org>  Mon, 30 Oct 2006 21:35:10 -0500

scala (2.1.5-2) unstable; urgency=low

  * Rearranged to have three packages instead of four.

 -- Lex Spoon <lex@debian.org>  Wed, 28 Jun 2006 17:16:02 +0200

scala (2.1.5-1) unstable; urgency=low

  * upstream update.

 -- Lex Spoon <lex@debian.org>  Sun, 9 Jun 2006 14:45:26 +0100

scala (2.1.1-1) unstable; urgency=low

  * Initial Release.

 -- Lex Spoon <lex@debian.org>  Sun, 26 Mar 2006 14:45:26 +0100



src/manual/scala/man1/scalac.scala
/* NSC -- new Scala compiler
 * Copyright 2005-2008 LAMP/EPFL
 * @author Stephane Micheloud
 */
//$Id$

package scala.man1

/**
 *  @author Stephane Micheloud
 */
object scalac extends Command {
  import _root_.scala.tools.docutil.ManPage._

  protected def cn = new Error().getStackTrace()(0).getClassName()

  val name = Section("NAME",

    MBold(command) & " " & NDash & " Compiler for the " &
    Link("Scala 2", "http://scala-lang.org/") & " language")

  val synopsis = Section("SYNOPSIS",

    CmdLine(" [ " & Argument("options") & " ] " &
            Argument("source files")))

  val parameters = Section("PARAMETERS",

    DefinitionList(
      Definition(
        Mono(Argument("options")),
        "Command line options. See " & Link(Bold("OPTIONS"), "#options") &
        " below."),
      Definition(
        Mono(Argument("source files")),
        "One or more source files to be compiled (such as " &
        Mono("MyClass.scala") & ").")))

  val description = Section("DESCRIPTION",

    "The " & MBold(command) & " tool reads class and object definitions, " &
    "written in the Scala programming language, and compiles them into " &
    "bytecode class files.",

    "By default, the compiler puts each class file in the same directory " &
    "as its source file. You can specify a separate destination directory " &
    "with -d (see " & Link(Bold("OPTIONS"), "#options") & ", below).")

  val options = Section("OPTIONS",

    "The compiler has a set of standard options that are supported on the " &
    "current development environment and will be supported in future " &
    "releases. An additional set of non-standard options are specific to " &
    "the current virtual machine implementation and are subject to change " &
    "in the future.  Non-standard options begin with " & Bold("-X") & ".",

    Section("Standard Options",
      DefinitionList(
        Definition(
          CmdOption("g:{none,source,line,vars,notc}"),
          SeqPara(
            Mono("\"none\"") & " generates no debugging info,",
            Mono("\"source\"") & " generates only the source file attribute,",
            Mono("\"line\"") & " generates source and line number information,",
            Mono("\"vars\"") & " generates source, line number and local " &
            "variable information,",
            Mono("\"notc\"") & " generates all of the above and " &
            Italic("will not") & " perform tail call optimization.")),
        Definition(
          CmdOption("nowarn"),
          "Generate no warnings"),
        Definition(
          CmdOption("verbose"),
          "Output messages about what the compiler is doing"),
        Definition(
          CmdOption("deprecation"),
          SeqPara(
            "Indicate whether source should be compiled with deprecation " &
            "information; defaults to " & Mono("off") & " (" &
            "accepted values are: " & Mono("on") & ", " & Mono("off") &
            ", " & Mono("yes") & " and " & Mono("no") & ")",
            "Available since Scala version 2.2.1")),
        Definition(
          CmdOption("unchecked"),
          SeqPara(
            "Enable detailed unchecked warnings",
            "Non variable type-arguments in type patterns are unchecked " &
            "since they are eliminated by erasure",
            "Available since Scala version 2.3.0")),
        Definition(
          CmdOption("classpath", Argument("path")),
          SeqPara(
            "Specify where to find user class files (on Unix-based systems " &
            "a colon-separated list of paths, on Windows-based systems, a " &
            "semicolon-separate list of paths). This does not override the " &
            "built-in (" & Mono("\"boot\"") & ") search path.",
            "The default class path is the current directory. Setting the " &
            Mono("CLASSPATH") & " variable or using the " & Mono("-classpath") & " " &
            "command-line option overrides that default, so if you want to " &
            "include the current directory in the search path, you must " &
            "include " & Mono("\".\"") & " in the new settings.")),
        Definition(
          CmdOption("sourcepath", Argument("path")),
          "Specify where to find input source files."),
        Definition(
          CmdOption("bootclasspath", Argument("path")),
          "Override location of bootstrap class files (where to find the " &
          "standard built-in classes, such as \"" & Mono("scala.List") & "\")."),
        Definition(
          CmdOption("extdirs", Argument("dirs")),
          "Override location of installed extensions."),
        Definition(
          CmdOption("d", Argument("directory")),
          "Specify where to place generated class files."),
        Definition(
          CmdOption("encoding", Argument("encoding")),
          SeqPara(
            "Specify character encoding used by source files.",
            "The default value is platform-specific (Linux: " & Mono("\"UTF8\"") &
            ", Windows: " & Mono("\"Cp1252\"") & "). Executing the following " &
            "code in the Scala interpreter will return the default value " &
            "on your system:",
            MBold("    scala> ") &
            Mono("new java.io.InputStreamReader(System.in).getEncoding"))),
        Definition(
          CmdOption("target:", Argument("target")),
          SeqPara(
            "Specify which backend to use (" & Mono("jvm-1.5,jvm-1.4," &
            "msil,cldc") & ").",
            "The default value is " & Mono("\"jvm-1.5\"") & " (was " &
            Mono("\"jvm-1.4\"") & " up to Scala version 2.6.1).")),
        Definition(
          CmdOption("print"),
          "Print program with all Scala-specific features removed"
        ),
        Definition(
          CmdOption("optimise"),
          "Generates faster bytecode by applying optimisations to the program"
        ),
        Definition(
          CmdOption("explaintypes"),
          "Explain type errors in more detail."),
        Definition(
          CmdOption("uniqid"),
          "Print identifiers with unique names (debugging option)."),
        Definition(
          CmdOption("version"),
          "Print product version and exit."),
        Definition(
          /*CmdOption("?") & "| " &*/ CmdOption("help"),
          "Print a synopsis of standard options."))),

    Section("Advanced Options",
      DefinitionList(
        Definition(
          CmdOption("Xassem", Argument("file")),
          "Name of the output assembly (only relevant with -target:msil)"),
        Definition(
          CmdOption("Xassem-path", Argument("path")),
          "List of assemblies referenced by the program (only relevant with -target:msil)"),
        Definition(
          CmdOption("Xcheck-null"),
          "Emit warning on selection of nullable reference"),
        Definition(
          CmdOption("Xdisable-assertions"),
          "Generate no assertions and assumptions"),
        Definition(
          CmdOption("Xexperimental"),
          "enable experimental extensions"),
        Definition(
          CmdOption("Xno-uescape"),
          "Disable handling of " & BSlash & "u unicode escapes"),
        Definition(
          CmdOption("Xplug-types"),
          "Parse but ignore annotations in more locations"),
        Definition(
          CmdOption("Xplugin:", Argument("file")),
          "Load a plugin from a file"),
        Definition(
          CmdOption("Xplugin-disable:", Argument("plugin")),
          "Disable a plugin"),
        Definition(
          CmdOption("Xplugin-list"),
          "Print a synopsis of loaded plugins"),
        Definition(
          CmdOption("Xplugin-opt:", Argument("plugin:opt")),
          "Pass an option to a plugin"),
        Definition(
          CmdOption("Xplugin-require:", Argument("plugin")),
          "Abort unless a plugin is available"),
        Definition(
          CmdOption("Xprint:", Argument("phases")),
          "Print out program after " & Argument("phases") & " (see below)."),
        Definition(
          CmdOption("Xprint-pos"),
          "Print tree positions (as offsets)"),
        Definition(
          CmdOption("Xprint-types"),
          "Print tree types (debugging option)."),
        Definition(
          CmdOption("Xprompt"),
          "Display a prompt after each error (debugging option)."),
        Definition(
          CmdOption("Xresident"),
          "Compiler stays resident, files to compile are read from standard " &
          "input."),
        Definition(
          CmdOption("Xshow-class", Argument("class")),
          "Show class info."),
        Definition(
          CmdOption("Xshow-object", Argument("object")),
          "Show object info."),
        Definition(
          CmdOption("Xshow-phases"),
          "Print a synopsis of compiler phases."),
        Definition(
          CmdOption("Xsource-reader", Argument("classname")),
          "Specify a custom method for reading source files."),
        Definition(
          CmdOption("Xscript", Argument("object")),
          "Compile as a script, wrapping the code into object.main().")
      )
    ),

    Section("Compilation Phases",
      DefinitionList(
        Definition(
          MItalic("initial"),
          "initializing compiler"),
        Definition(
          MItalic("parse"),
          "parse source files"),
        Definition(
          MItalic("namer"),
          "create symbols"),
        Definition(
          MItalic("analyze"),
          "name and type analysis"),
        Definition(
          MItalic("refcheck"),
          "reference checking"),
        Definition(
          MItalic("uncurry"),
          "uncurry function types and applications"),
        Definition(
          MItalic("transmatch"),
          "translate match expressions"),
        Definition(
          MItalic("lambdalift"),
          "lambda lifter"),
        Definition(
          MItalic("typesasvalues"),
          "represent types as values"),
        Definition(
          MItalic("addaccessors"),
          "add accessors for constructor arguments"),
        Definition(
          MItalic("explicitouterclasses"),
          "make links from inner classes to enclosing one explicit"),
        Definition(
          MItalic("addconstructors"),
          "add explicit constructor for each class"),
        Definition(
          MItalic("tailcall"),
          "add tail-calls"),
        Definition(
          MItalic("wholeprog"),
          "perform whole program analysis"),
        Definition(
          MItalic("addinterfaces"),
          "add one interface per class"),
        Definition(
          MItalic("expandmixins"),
          "expand mixins by code copying"),
        Definition(
          MItalic("boxing"),
          "makes boxing explicit"),
        Definition(
          MItalic("erasure"),
          "type eraser"),
        Definition(
          MItalic("icode"),
          "generate icode"),
        Definition(
          MItalic("codegen"),
          "enable code generation"),
        Definition(
          MItalic("terminal"),
          "compilation terminated"),
        Definition(
          MItalic("all"),
          "matches all phases"))))

  val environment = Section("ENVIRONMENT",

    DefinitionList(
      Definition(
        MBold("JAVACMD"),
        "Specify the " & MBold("java") & " command to be used " &
        "for running the Scala code.  Arguments may be specified " &
        "as part of the environment variable; spaces, quotation marks, " &
        "etc., will be passed directly to the shell for expansion."),
      Definition(
        MBold("JAVA_OPTS"),
        SeqPara(
          "Specify the options to be passed to the " & MBold("java") &
          " command defined by " & MBold("JAVACMD") & ".",

          "With Java 1.5 (or newer) one may for example configure the " &
          "memory usage of the JVM as follows: " &
          Mono("JAVA_OPTS=\"-Xmx512M -Xms16M -Xss16M\""),

          "With " & Link("GNU Java", "http://gcc.gnu.org/java/") & " one " &
          "may configure the memory usage of the GIJ as follows: " &
          Mono("JAVA_OPTS=\"--mx512m --ms16m\"")
        ))))

  val examples = Section("EXAMPLES",

    DefinitionList(
      Definition(
        "Compile a Scala program to the current directory",
        CmdLine("HelloWorld")),
      Definition(
        "Compile a Scala program to the destination directory " &
        MBold("classes"),
        CmdLine(CmdOption("d", "classes") & "HelloWorld.scala")),
     Definition(
        "Compile a Scala program using a user-defined " & MBold("java") & " " &
        "command",
        MBold("env JAVACMD") & Mono("=/usr/local/bin/cacao ") &
        CmdLine(CmdOption("d", "classes") & "HelloWorld.scala")),
      Definition(
        "Compile all Scala files found in the source directory " &
        MBold("src") & " to the destination directory " &
        MBold("classes"),
        CmdLine(CmdOption("d", "classes") & "src/*.scala"))))

  val exitStatus = Section("EXIT STATUS",

    MBold(command) & " returns a zero exist status if it succeeds to " &
    "compile the specified input files. Non zero is returned in case " &
    "of failure.")

  val seeAlso = Section("SEE ALSO",

    Link(Bold("fsc") & "(1)", "fsc.html") & ", " &
    Link(Bold("sbaz") & "(1)", "sbaz.html") & ", " &
    Link(Bold("scala") & "(1)", "scala.html") & ", " &
    Link(Bold("scaladoc") & "(1)", "scaladoc.html") & ", " &
    Link(Bold("scalap") & "(1)", "scalap.html"))

  def manpage = new Document {
    title = command
    date = lastModified // e.g. "June 8, 2006"
    author = "Stephane Micheloud & LAMP"
    version = "0.4"
    sections = List(
      name,
      synopsis,
      parameters,
      options,
      environment,
      examples,
      exitStatus,
      authors,
      bugs,
      copyright,
      seeAlso)
  }
}
